### PR TITLE
[ET-1376] stripe general settings

### PR DESCRIPTION
**Pull request by codingmusician**
### 🎫 Ticket

[ET-1376] 

### 🗒️ Description

This ticket is basically putting the group headers into place. I did a little work for [ET-1357] here, too, but I'll complete that work in a different PR. 

### 🎥 Artifacts 

https://youtu.be/S6u28KzRs20

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1376]: https://theeventscalendar.atlassian.net/browse/ET-1376?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ET-1357]: https://theeventscalendar.atlassian.net/browse/ET-1357?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bordoni — DISMISSED** [review]
So far this looks good.

**Review by juanfra — COMMENTED** [review]
So far looks good! We'll probably need to update snapshots. It'd be interesting, given that we're replicating the connection section for Stripe, that we also include tests for these new templates as well.

File: src/Tickets/Commerce/Gateways/Stripe/REST/Publishable_Key_Endpoint.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	 *
+	 * @since TBD
+	 */
+	public function register() {
+		$namespace     = tribe( 'tickets.rest-v1.main' )->get_events_route_namespace();
+		$documentation = tribe( 'tickets.rest-v1.endpoints.documentation' );
+
+		register_rest_route(
+			$namespace,
+			$this->get_endpoint_path(),
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/REST/Publishable_Key_Endpoint.php:45]: I dont think we need this file anymore. @moraleida confirm?

File: src/admin-views/settings/tickets-commerce/gateways/status.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+$classes = [
+	'tec-tickets__admin-settings-tickets-commerce-gateways-item-status',
+	'tec-tickets__admin-settings-tickets-commerce-gateways-item-status--enabled' => $gateway->is_enabled() && $gateway->is_active(),
+];
+
+?>
+<div <?php tribe_classes( $classes ); ?>>
+	<?php echo $gateway->get_status_text(); ?>
```
Comment (juanfra) [inline src/admin-views/settings/tickets-commerce/gateways/status.php:32]:
```suggestion
	<?php echo $gateway->get_status_text(); // phpcs:ignore ?>
```

File: src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss [hunk 3]
```diff
@@ -496,6 +506,36 @@
...
+	&:active {
+		background-color: var(--tec-color-button-primary-active);
+	}
+}
+
+.tec-tickets__admin-settings-tickets-commerce-gateways-item-button-link--active {
+	background-color: #fff;
+	border-color: #3c434a;
+	border-style: solid;
+	border-width: 1px;
```
Comment (juanfra) [inline src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss:525]: We have a variable for this border with.

File: src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss [hunk 4]
```diff
@@ -496,6 +506,36 @@
...
+	color: #3c434a;
+
+	&:active,
+	&:hover {
+		background-color: var(--tec-color-button-primary-hover);
+		border-color: var(--tec-color-button-primary-hover);
+		color: #fff;
```
Comment (juanfra) [inline src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss:532]: We have a variable for white.

File: src/views/v2/commerce/checkout/cart.php [hunk 5]
```diff
@@ -13,20 +13,21 @@
...
+ * @var string           $login_url        [Global] The site's login URL.
+ * @var string           $registration_url [Global] The site's registration URL.
+ * @var bool             $is_tec_active    [Global] Whether `The Events Calendar` is active or not.
+ * @var int              $section          Which Section that we are going to render for this table.
+ * @var array[]          $gateways         [Global] An array with the gateways.
+ * @var int              $gateways_active  [Global] The number of active gateways.
  */
 
 $post = get_post( $section );
+$user = wp_get_current_user();
```
Comment (juanfra) [inline src/views/v2/commerce/checkout/cart.php:30]: Let's move the `$user` to billing details, as we don't use it here. We should get that only if we're running the template.

File: src/views/v2/commerce/checkout/cart.php [hunk 6]
```diff
@@ -35,7 +36,7 @@
 
 	<?php $this->template( 'checkout/cart/items', [ 'post' => $post ] ); ?>
 
-	<?php $this->template( 'checkout/cart/billing_identification' ); ?>
+	<?php $this->template( 'checkout/cart/billing-details', [ 'user' => $user ] ); ?>
```
Comment (juanfra) [inline src/views/v2/commerce/checkout/cart.php:39]: The billing details shouldn't be part of the cart, we should move this outside of it. This comment is a reminder for the FE audit. Shouldn't be addressed now.

File: src/views/v2/commerce/checkout/cart/billing-details.php [hunk 7]
```diff
@@ -12,33 +12,31 @@
...
+$username  = ! empty( $user->display_name ) ? $user->display_name : '';
+$useremail = ! empty( $user->user_email ) ? $user->user_email : '';
 
 ?>
 
 <div id="tec-tc-gateway-stripe-billing-identification">
 	<h4><?php esc_html_e( 'Billing Information', 'event-tickets' ); ?></h4>
 	<span id="tec-tc-gateway-stripe-billing-first-name">
-		<input type="text" value="<?php echo esc_attr( $username ); ?>" placeholder="First Name"/>
+		<input type="text" value="<?php echo esc_attr( $username ); ?>"
```
Comment (juanfra) [inline src/views/v2/commerce/checkout/cart/billing-details.php:31]: We're using `$username` for firs and last name.

**Review by juanfra — COMMENTED** [review]
Looking great! A few comments

File: src/Tickets/Commerce/Gateways/Stripe/REST/Publishable_Key_Endpoint.php (hunk 1 above)
Comment (moraleida) [reply src/Tickets/Commerce/Gateways/Stripe/REST/Publishable_Key_Endpoint.php:45]: Correct, this file was removed in my branch

File: src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss (hunk 3 above)
Comment (codingmusician) [reply src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss:525]: If we do, I don't see it. Which file are you looking at?

File: src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss (hunk 4 above)
Comment (codingmusician) [reply src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss:532]: I do see a variable for a background color, but not for text color. Should I use the BG color for text?

File: src/views/v2/commerce/checkout/cart.php (hunk 5 above)
Comment (codingmusician) [reply src/views/v2/commerce/checkout/cart.php:30]: Hmm. This must have been something that was done in one of Ricardo's tickets. I'll move it when I get to the checkout tickets. I'll add a `@todo`

File: src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss (hunk 3 above)
Comment (juanfra) [reply src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss:525]: @codingmusician You are right, we don't! I would have bet that we had one here: https://github.com/the-events-calendar/tribe-common-styles/blob/main/variables/_borders.pcss

File: src/views/v2/commerce/checkout/cart/billing-details.php (hunk 7 above)
Comment (codingmusician) [reply src/views/v2/commerce/checkout/cart/billing-details.php:31]: Will add a `@todo` to address this in the front end tickets.

File: src/views/v2/commerce/checkout/cart.php (hunk 6 above)
Comment (codingmusician) [reply src/views/v2/commerce/checkout/cart.php:39]: Will add a `@todo` for this.

File: src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss (hunk 4 above)
Comment (juanfra) [reply src/resources/postcss/tickets-admin/settings/_tickets-commerce.pcss:532]: @codingmusician we do use it in [some places](https://github.com/the-events-calendar/tribe-common/blob/master/src/resources/postcss/components/full/buttons/_border.pcss#L27), but we can keep the hex value. I'll leave it to you.

**Review by juanfra — APPROVED** [review]

**Pull request merged**